azext_edge/tests/edge/orchestration/resources/connector/opcua/test_opcua_certs_unit.py

```diff
@@ -0,0 +1,913 @@
+                method=responses.PUT,
+                url=get_secret_endpoint(keyvault_name="mock-keyvault", secret_name=secret_name),
+                json={},
+                status=200,
+                content_type="application/json",
+            )
+
+            # get opcua spc
+            mocked_responses.add(
+                method=responses.GET,
+                url=get_spc_endpoint(spc_name=OPCUA_SPC_NAME, resource_group_name=rg_name),
+                json=issuer_list_spc,
+                status=200,
+                content_type="application/json",
+            )
+
+            if issuer_list_spc:
+                # set opcua spc
+                mocked_responses.add(
+                    method=responses.PUT,
+                    url=get_spc_endpoint(spc_name=OPCUA_SPC_NAME, resource_group_name=rg_name),
+                    json={},
+                    status=200,
+                    content_type="application/json",
+                )
+
+            if issuer_list_secretsync:
+                # set opcua secretsync
+                mocked_responses.add(
+                    method=responses.PUT,
+                    url=get_secretsync_endpoint(
+                        secretsync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME, resource_group_name=rg_name
+                    ),
+                    json=expected_secret_sync,
+                    status=200,
+                    content_type="application/json",
+                )
+
+    result = None
+
+    try:
+        result = add_connector_opcua_issuer(
+            cmd=mocked_cmd,
+            instance_name=instance_name,
+            resource_group=rg_name,
+            file=file_name,
+            secret_name=secret_name,
+        )
+    except Exception:
+        if not issuer_list_spc:
+            assert (
+                mocked_logger.warning.call_args[0][0] == f"Azure Key Vault Secret Provider Class {OPCUA_SPC_NAME} "
+                "not found, creating new one..."
+            )
+            return
+
+        if not issuer_list_secretsync:
+            assert (
+                mocked_logger.warning.call_args[0][0] == f"Secret Sync {OPCUA_ISSUER_LIST_SECRET_SYNC_NAME} "
+                "not found, creating new one..."
+            )
+            return
+
+    if not expected_resources_map["custom locations"]:
+        assert (
+            mocked_logger.error.call_args[0][0] == f"Secret sync is not enabled for the instance {instance_name}. "
+            "Please enable secret sync before adding certificate."
+        )
+        return
+
+    if not matched_names and file_name.endswith(".crl"):
+        file_name = os.path.basename(file_name)
+        assert (
+            mocked_logger.error.call_args[0][0] == f"Cannot add .crl {file_name} without "
+            "corresponding .crt or .der file."
+        )
+        return
+
+    if result:
+        assert result == expected_secret_sync
+
+
+@pytest.mark.parametrize("file_content", [b"\x00\x01\x02\x03"])
```

Comment:
if it is only one value with no plans to add more, just make it a variable in the test body